src/axom/primal/tests/primal_tetrahedron.cpp

```diff
@@ -446,35 +503,97 @@ TEST_F(TetrahedronTest, tetrahedron_roundtrip_bary_to_physical)
           EXPECT_NEAR(0., primal::squared_distance(p_exp[i], b2p), EPS);
 
           RPoint p2b = tet.physToBarycentric(b2p);
+          EXPECT_NEAR(p2b[0] + p2b[1] + p2b[2] + p2b[3], 1., EPS);
           EXPECT_NEAR(0., primal::squared_distance(b_in[i], p2b), EPS);
         }
       }
 
       // test tet barycenters
       {
-        RPoint b_in[1] = {RPoint {.25, .25, .25, .25}};
+        RPoint b_in {.25, .25, .25, .25};
+        QPoint p_exp(.25 * (tet[0].array() + tet[1].array() + tet[2].array() + tet[3].array()));
 
-        QPoint p_exp[1] = {
-          QPoint(.25 * (tet[0].array() + tet[1].array() + tet[2].array() + tet[3].array()))};
+        QPoint b2p = tet.baryToPhysical(b_in);
+        EXPECT_NEAR(0., primal::squared_distance(p_exp, b2p), EPS);
 
-        for(int i = 0; i < 1; ++i)
+        RPoint p2b = tet.physToBarycentric(b2p);
+        EXPECT_NEAR(p2b[0] + p2b[1] + p2b[2] + p2b[3], 1., EPS);
+        EXPECT_NEAR(0., primal::squared_distance(b_in, p2b), EPS);
+      }
+
+      // test outside points (several permutations of barycentric coordinates)
+      {
+        std::vector<double> coords {-.4, 2., .2};
+        coords.push_back(1. - coords[0] - coords[1] - coords[2]);
+
+        do
         {
-          QPoint b2p = tet.baryToPhysical(b_in[i]);
-          EXPECT_NEAR(0., primal::squared_distance(p_exp[i], b2p), EPS);
+          EXPECT_NEAR(coords[0] + coords[1] + coords[2] + coords[3], 1., EPS);
+
+          RPoint b_in(coords.data());
+          QPoint p_exp(b_in[0] * tet[0].array() + b_in[1] * tet[1].array() +
+                       b_in[2] * tet[2].array() + b_in[3] * tet[3].array());
+
+          QPoint b2p = tet.baryToPhysical(b_in);
+          EXPECT_NEAR(0., primal::squared_distance(p_exp, b2p), EPS);
 
           RPoint p2b = tet.physToBarycentric(b2p);
-          EXPECT_NEAR(0., primal::squared_distance(b_in[i], p2b), EPS);
-        }
+          EXPECT_NEAR(p2b[0] + p2b[1] + p2b[2] + p2b[3], 1., EPS);
+          EXPECT_NEAR(0., primal::squared_distance(b_in, p2b), EPS);
+
+        } while(std::next_permutation(coords.begin(), coords.end()));
       }
     }
   }
 }
 
+TEST_F(TetrahedronTest, tetrahedron_containment)
```

Comment:
This was a very concise but thorough way to set up the containment testing. Nicely done!